packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -264,9 +346,20 @@ Static group authorization should perform as expected.`
                 }
                 // We also need to add a stack mapping so that this resolver is added to the model stack.
                 ctx.mapResourceToStack(typeName, resolverResourceId)
-                resolver = this.resources.blankResolver(typeName, fieldName)
+                resolver = this.resources.blankResolver(typeName, field.name.value)
             }
             const authExpression = this.authorizationExpressionOnSingleObject(rules, 'ctx.source')
+            if (protectPrivateFields) {
+                if (field.type.kind === Kind.NON_NULL_TYPE) {
+                    throw new InvalidDirectiveError(`\nPer-field auth on the required field ${field.name.value} is not supported with subscriptions.
+Either make the field optional, set auth on the object and not the field, or disable subscriptions for the object (setting level to OFF or PUBLIC)\n`)
+                }
+                // add operation to queryField
+                this.protectMutations(ctx, typeName, 'mutation')
```

Comment:
Don't use ```'mutation'``` literal here, use ```ctx.getMutationTypeName```. What other value this parameter can have? Because the called function name suggests only mutations are supported. The comment above suggests queryField, which is not correct.